lib/ansible/modules/web_infrastructure/pmp.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2017, Bernat Mut <bernat.mut@aireuropa.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+module: pmp
+version_added: "2.5"
+short_description: Create and delete resources in Password Manager Pro (PMP)
+description:
+  - This allows to create and delete a resource with an account inside Password Manager Pro (PMP).
+
+options:
+  uri:
+    required: true
+    description:
+      - Base URI for the PMP instance.
+  port:
+    required: false
+    default:  7272
+    description:
+      - Base URI for the PMP instance.
+  state:
+    required: true
+    choices: [ present, absent ]
+    description:
+      -  Create or delete resource.
+
+  token:
+    required: true
+    description:
+      - Token of the API user that performs the action (must have enough privileges)
+
+  group:
+    required: false
+    description:
+      - If exists set the main group of the host.
+
+  resource_name:
+    required: true
+    description:
+     - The name of the resource that will be created.
+
+  account_name:
+    required: true
+    description:
+     - User associated with the created resource.
+
+  password:
+    required: true
+    description:
+     - Password associated with the user_name.
+
+  owner:
+    required: true
+    description:
+     - Change the default owner, must be an existing user.
+
+  use_proxy:
+    required: false
+    description:
+      - Set if open_url must use proxy
+  validate_certs:
+    required: false
+    description:
+      - Set if open_url must validate certificates
+
+author: "Bernat Mut (@berni69)"
+"""
+
+EXAMPLES = """
+- name: Delete host
+  pmp:
+    uri: '{{ pmp_server }}'
+    port: '{{ pmp_port }}'
+    token: '{{ user_token }}'
+    state: 'absent'
+    resource_name: '{{ server_name }}'
+    use_proxy: False
+    validate_certs: False
+  register: pmp_out
+  delegate_to: localhost
+
+- name: Create server host
+  pmp:
+    uri: '{{ pmp_server }}'
+    port: '{{ pmp_port }}'
+    state: 'present'
+    token: '{{ user_token }}'
+    resource_name: '{{ server_name }}'
+    group: '{{ my_group}}'
+    account_name: '{{ user_name }}'
+    password: '{{ password }}'
+    owner: 'DOMAIN\\USER'
+"""
+
+RETURN = '''
+status:
+    description: status of action executed
+    returned: success
+    type: string
+    sample: 'OK'
+'''
+
+import json
+import urllib
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url
+
+
+class PasswordManagerPro:
+    def format(self, strr):
+        for key, value in self.data.items():
+            strr = strr.replace('{' + key + '}', value)
```

Comment:
It's safer to quote unvalidated strings with, e.g., [`urllib.quote`](https://docs.python.org/2.7/library/urllib.html?#urllib.quote) before concatenating them into a URL.